hazelcast/src/test/java/com/hazelcast/client/ClientClusterRestartEventTest.java

```diff
@@ -137,6 +139,7 @@ public void memberRemoved(MembershipEvent membershipEvent) {
             }
         });
 
+        changeClusterStateEventually(instance1, ClusterState.PASSIVE);
```

Comment:
I am a bit confused that in the Cluster.shutdown doc it says: "Changes state of the cluster to the {@link ClusterState#PASSIVE} transactionally", hence us adding a code here to change cluster state looks like a dirty workaround.

From the problem statement I see that you mention: "don't have up-to-date partition table version" , then it may be more natural to wait for the partition table update and finish instead of this fix, what do you think?

Reply:
Note that the javadoc of `Cluster#shutdown` also says:
```
* @throws IllegalStateException if member-list changes during the transaction
*                               or there are ongoing/pending migration operations
*                               or shutdown process times out
```
I discussed this with @mdogan and apparently this is a deliberate design decision on the shutdown level - it can fail for a number of reasons and the issue with the partition table version is just one of them. We concluded it's not worth trying to fix it in shutdown itself.

Reply:
> The test `ClientClusterStateTest` also uses a similar cluster shutdown, do we need it at that test as well?

`changeClusterStateEventually` is used for a different purpose in that test - to test behavior in various cluster states). In our case, it's used more as a simple tool to wait that the cluster is in the expected state (partition tables up to date etc.) before we call shutdown.

Reply:
It is not possible to implement this test with the same work around for non-java clients. Hence we decided to do the work around as this:
```
        final Cluster cluster = instance1.getCluster(); 
        assertTrueEventually(new AssertTask() {
            @Override
            public void run() throws Exception {
                cluster.shutdown();
            }
        });
```